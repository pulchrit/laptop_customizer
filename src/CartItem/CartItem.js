import React from 'react';
import "../App.css";

class CartItem extends React.Component {

    render() {
        
        return (
            <div className="summary__option" key={this.props.selectedFeatureName}>
                
                <div className="summary__option__label">
                    {this.props.selectedFeatureName}
                </div>
                
                <div className="summary__option__value">
                    {this.props.selectedState[this.props.selectedFeatureName].name}
                </div>
                
                <div className="summary__option__cost">
                    { new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
                    .format(this.props.selectedState[this.props.selectedFeatureName].cost) }
                </div>
            </div>

        );
    }
}

export default CartItem;