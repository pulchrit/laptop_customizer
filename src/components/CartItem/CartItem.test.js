import React from 'react';
import ReactDOM from 'react-dom';
import CartItem from "./CartItem";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("CartItem component", () => {
    
    const selected = {
        Processor: {
            name: '17th Generation Intel Core HB (7 Core with donut spare)',
            cost: 700
        },
        "Operating System": {
            name: 'Ubuntu Linux 16.04',
            cost: 200
        },
        "Video Card": {
            name: 'Toyota Corolla 1.5v',
            cost: 1150.98
        },
        Display: {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
        }
    };

    const selectedFeatureName = "Display";

    // smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CartItem />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // snapshot test
    it('renders Display cart item', () => {
        const wrapper = shallow(<CartItem 
            key={selectedFeatureName}
            selectedFeatureName={selectedFeatureName}
            selectedState={selected}
        />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});


{/* This is how CartItem is called:
    <CartItem 
                    key={key}
                    selectedFeatureName={key}
                    selectedState={this.props.selectedState}
                /> */}