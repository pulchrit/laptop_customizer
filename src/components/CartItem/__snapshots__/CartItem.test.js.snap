// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component renders Display cart item 1`] = `
<div
  className="summary__option"
  key="Display"
>
  <div
    className="summary__option__label"
  >
    Display
  </div>
  <div
    className="summary__option__value"
  >
    15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
  </div>
  <div
    className="summary__option__cost"
  >
    $1,500.00
  </div>
</div>
`;
